sentry/src/main/java/io/sentry/SentryTransaction.java

```diff
@@ -0,0 +1,218 @@
+    return this.startChild(getSpanId());
+  }
+
+  /**
+   * Starts a child Span with given trace id and parent span id.
+   *
+   * @param parentSpanId - parent span id
+   * @return a new transaction span
+   */
+  Span startChild(final @NotNull SpanId parentSpanId) {
+    Objects.requireNonNull(parentSpanId, "parentSpanId is required");
+    final Span span = new Span(getTraceId(), parentSpanId, this);
+    this.spans.add(span);
+    return span;
+  }
+
+  @Override
+  public SentryTraceHeader toSentryTrace() {
+    return new SentryTraceHeader(getTraceId(), getSpanId(), isSampled());
+  }
+
+  @NotNull
+  SpanId getSpanId() {
+    return getContexts().getTrace().getSpanId();
+  }
+
+  @NotNull
+  SentryId getTraceId() {
+    return getContexts().getTrace().getTraceId();
+  }
+
+  @Nullable
+  Boolean isSampled() {
+    return getContexts().getTrace().getSampled();
+  }
+
+  @Override
+  public void finish() {
+    this.timestamp = DateUtils.getCurrentDateTime();
+    this.hub.captureTransaction(this, null);
+  }
+
+  /**
+   * Sets transaction operation.
+   *
+   * @param op - operation
+   */
+  @Override
+  public void setOp(@Nullable String op) {
+    this.getContexts().getTrace().setOp(op);
+  }
+
+  /**
+   * Sets transaction description.
+   *
+   * @param description - the description.
+   */
+  @Override
+  public void setDescription(@Nullable String description) {
+    this.getContexts().getTrace().setDescription(description);
+  }
+
+  public @Nullable String getDescription() {
+    return this.getContexts().getTrace().getDescription();
+  }
+
+  @Override
+  public @NotNull SpanContext getSpanContext() {
+    return this.getContexts().getTrace();
+  }
+
+  /**
+   * Sets transaction status.
+   *
+   * @param spanStatus - the status
+   */
+  @Override
+  public void setStatus(@Nullable SpanStatus spanStatus) {
+    this.getContexts().getTrace().setStatus(spanStatus);
+  }
+
+  /**
+   * Returns the transaction name.
+   *
+   * @return transaction name.
+   */
+  @Nullable
+  String getTransaction() {
+    return transaction;
+  }
+
+  @NotNull
+  Date getStartTimestamp() {
+    return startTimestamp;
+  }
+
+  @Nullable
+  Date getTimestamp() {
+    return timestamp;
+  }
+
+  @NotNull
+  List<Span> getSpans() {
+    return spans;
+  }
+
+  /** @return the latest span that is not finished or null if not found. */
+  @Nullable
+  Span getLatestActiveSpan() {
+    final List<Span> spans = new ArrayList<>(this.spans);
+    for (int i = spans.size() - 1; i >= 0; i--) {
+      if (!spans.get(i).isFinished()) {
+        return spans.get(i);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Gets the span context for the span that was active while the throwable given by parameter was
+   * thrown.
+   *
+   * @param throwable - the throwable
+   * @return span context or {@code null} if no corresponding span context found.
+   */
+  public SpanContext getSpanContext(final @NotNull Throwable throwable) {
```

Comment:
H: Just realized that this approach will not work once we flush individual spans (i.e: Don't accumulate all spans in all open transactions on the server, mainly due to the memory pressure but instead flush them out as they close to allow batches to be flushed to the server)